src/main/webapp/app/modules/games/single-game/single-game.tsx

```diff
@@ -0,0 +1,128 @@
+import '../games.scss';
+
+import {RouteComponentProps} from "react-router-dom";
+import {connect} from "react-redux";
+import {IRootState} from "app/shared/reducers";
+import {getSingleGame} from "app/modules/games/games.reducer";
+import React, {useEffect, useState} from "react";
+import {Col, Container, DropdownItem, Row} from 'reactstrap';
+import {cleanText, formatCurrency} from "app/shared/util/render-utils";
+import {CartQuantityBox} from "app/modules/games/single-game/cart-quantity-box";
+import {CategoriesMechanics} from "app/modules/games/single-game/categories-mechanics";
+import {Carousel} from "app/modules/games/single-game/carousel";
+
+export interface ISingleGameProps extends StateProps, DispatchProps, RouteComponentProps<{ id: string }> {
+}
+
+export const SingleGame = (props: ISingleGameProps) => {
+  const [formattedCurrency, setFormattedCurrency] = useState("");
+  const [quantityDropdown, setQuantityDropdown] = useState([]);
+  const [desiredQuantity, setDesiredQuantity] = useState(1);
+  const [checkoutDisabled, setCheckoutDisabled] = useState(false);
+
+  const createQuantityDropdown = (quantity: number) => {
+    const items = [];
+    if (quantity <= 10) {
```

Comment:
Instead of having separate for loops based on quantity, I think you should just create a variable to store the max quantity (which is either the quantity or 10) and then just use that in your for loop as the condition for when to stop

Reply:
Good point - updated 👍